test-fixtures/update.py

```diff
@@ -75,15 +104,28 @@ def default_stack():
         
         # remove pattern prefix from current_semver to compare with largest
         current_semver = current_semver.split(pattern)[1]
-
-        if current_semver >= default_semver:
-            print('Xcode version unchanged or more recent! aborting.')
+        
+        print("current_semver: {0}".format(current_semver))
+        print("latest_stable_semver: {0}".format(latest_stable_semver))
+        print("latest_semver: {0}".format(latest_semver))
+        
+        # if we use an edge stack currently, see if there is a new stable version
+        if '-edge' in current_semver:
+            # if there is a stable stack of the same version or later, use stable stack.
+            # otherwise do nothing
+            if latest_stable_semver >= current_semver.replace('-edge', ''):
+                print("Stable version of Bitrise stack available: {0} ... updating bitrise.yml!".format(latest_stable_semver))
+                write_to_bitrise_yml(y, latest_stable_semver)
+            else:
+                print('No stable version of Bitrise stack available! aborting.')
+        # if we are using a stable stack, see if there is a new stable version
+        elif current_semver < latest_stable_semver:
+            print('New stable Bitrise stack available: {0} ... updating bitrise.yml!'.format(latest_stable_semver))
+            write_to_bitrise_yml(y, latest_stable_semver)
+        # if a new edge version is available, update bitrise.yml
+        elif current_semver < latest_semver:
+            print('New edge version of Bitrise stack available: {0} ... updating bitrise.yml!'.format(latest_semver))
+            write_to_bitrise_yml(y, latest_semver)
+        # we're using the absolutely latest version available
         else:
-            print('New Xcode version available: {0} ... updating bitrise.yml!'.format(default_semver))
-            # add prefix pattern back to be recognizable by bitrise
-            # as a valid stack value
-            y['workflows'][WORKFLOW]['meta']['bitrise.io']['stack'] = '{0}{1}'.format(pattern, default_semver)
```

Comment:
Since there are more than one reason to update the yml file, I moved the snippet to `write_to_bitrise_yml()`.